### PR TITLE
transforms: Switch to kornia AugmentationSequential

### LINKED ISSUE #2002: Anomaly with RandomGrayScale tests
### Description

So I noticed something strange while running tests for grayscale. Passing sample through our `AugmentationSequential` is changing the value of `sample["image"]` to be equal to that of `output["image"]`.

### Steps to reproduce

In theory, the following change should allow the test to pass but it does not.

```diff
@pytest.mark.parametrize(
    "weights",
    [
        torch.tensor([1.0, 1.0, 1.0]),
        torch.tensor([0.299, 0.587, 0.114]),
        torch.tensor([1.0, 2.0, 3.0]),
    ],
)
def test_random_grayscale_sample(weights: Tensor, sample: dict[str, Tensor]) -> None:
    aug = AugmentationSequential(RandomGrayscale(weights, p=1, keepdim=True), data_keys=["image"])
+    from copy import deepcopy
+    tmp = deepcopy(sample)
    output = aug(sample)
-    assert output["image"].shape == sample["image"].shape
-    assert output["image"].sum() == sample["image"].sum()
+    assert output["image"].shape == tmp["image"].shape
+    assert output["image"].sum() == tmp["image"].sum()
    for i in range(1, 3):
        assert torch.allclose(output["image"][0], output["image"][i])
```

But this throws the following errors:

```
========================================================================= short test summary info ==========================================================================
FAILED tests/transforms/test_color.py::test_random_grayscale_sample[weights1] - assert tensor(985.9200) == tensor(1128.)
FAILED tests/transforms/test_color.py::test_random_grayscale_sample[weights2] - assert tensor(1384.) == tensor(1128.)
======================================================================= 2 failed, 4 passed in 2.35s ========================================================================
```

### Version

0.6.0.dev0 (abceea0)

### PR DESCRIPTION
**Pull request by ashnair1**
Switch to kornia's `AugmentationSequential` in `transforms` submodule

Closes #2002

### CODE REVIEW

**Comment by ashnair1** [comment]
Will need to push a fix to kornia to allow non-float dtypes.

**Comment by adamjstewart** [comment]
> Will need to push a fix to kornia to allow non-float dtypes.

Any progress on this?

**Comment by ashnair1** [comment]
> > Will need to push a fix to kornia to allow non-float dtypes.
> 
> Any progress on this?

kornia/kornia#2882 is nearly complete. Just needs another round of review.

**Comment by robmarkcole** [comment]
One issue with the switch - the data_key `boxes` in torchgeo is not an [recognised value](https://kornia.readthedocs.io/en/latest/augmentation.container.html#kornia.augmentation.container.AugmentationSequential) in kornia implementation, closest is bbox

**Comment by ashnair1** [comment]
> One issue with the switch - the data_key `boxes` in torchgeo is not an [recognised value](https://kornia.readthedocs.io/en/latest/augmentation.container.html#kornia.augmentation.container.AugmentationSequential) in kornia implementation, closest is bbox

This is planned to be implemented via #1978.

**Comment by johnnv1** [comment]
> [kornia/kornia#2882](https://github.com/kornia/kornia/pull/2882) is nearly complete. Just needs another round of review.


@ashnair1 your patch is now available at the latest California release (v0.7.3)

File: tests/transforms/test_color.py [hunk 1]
```diff
@@ -33,12 +34,14 @@ def batch() -> dict[str, Tensor]:
     ],
 )
 def test_random_grayscale_sample(weights: Tensor, sample: dict[str, Tensor]) -> None:
-    aug = AugmentationSequential(RandomGrayscale(weights, p=1), data_keys=['image'])
+    aug = K.AugmentationSequential(
+        RandomGrayscale(weights, p=1), keepdim=True, data_keys=None
+    )
+    aug.keepdim = True
```
Comment (adamjstewart) [inline tests/transforms/test_color.py:40]: Why is this line needed? Isn't this already set on the previous line?

**Review by adamjstewart — APPROVED** [review]

File: tests/transforms/test_color.py (hunk 1 above)
Comment (adamjstewart) [reply tests/transforms/test_color.py:40]: Gotcha. Can you add:
```suggestion
    # https://github.com/kornia/kornia/issues/2848
    aug.keepdim = True
```
Then we'll know why and eventually be able to remove this line.

**Review by adamjstewart — APPROVED** [review]
